### PR TITLE
fix: -Wunsafe-buffer-usage warning in didRegisterForRemoteNotificationsWithDeviceToken

### PR DESCRIPTION
**Pull request by ckerr**
#### Description of Change

Part 21 in a [series](https://github.com/electron/electron/pull/43477) to fix `-Wunsafe-buffer-usage` warnings in `shell/`.  This is similar to the fix in #44127 which added an NSData getter helper, so this PR moves that helper into a public header so that it can be used for this fix, too.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: none.

### CODE REVIEW

File: shell/browser/mac/electron_application_delegate.mm [hunk 1]
```diff
@@ -176,18 +178,10 @@ - (IBAction)newWindowForTab:(id)sender {
...
-    [token_string appendFormat:@"%02.2hhX", token_data[i]];
-  }
   // Resolve outstanding APNS promises created during registration attempts
-  electron::api::PushNotifications* push_notifications =
-      electron::api::PushNotifications::Get();
-  if (push_notifications) {
+  if (auto* push_notifications = electron::api::PushNotifications::Get()) {
     push_notifications->ResolveAPNSPromiseSetWithToken(
-        base::SysNSStringToUTF8(token_string));
+        base::HexEncode(electron::util::as_byte_span(deviceToken)));
```
Comment (ckerr) [inline shell/browser/mac/electron_application_delegate.mm:184]: (implementation note: the `appendFormat:@"%0.02hhX"`  loop has been replaced with [base::HexEncode](https://chromium.googlesource.com/chromium/src/+/main/base/strings/string_number_conversions.cc#135), which uses [AppendHexEncodedByte()](https://chromium.googlesource.com/chromium/src/+/main/base/strings/string_number_conversions.h#117) which has the same two-uppercase-hex-chars-per-byte behavior as the `%02.2hhX` formatting call)

File: filenames.gni [hunk 2]
```diff
@@ -669,6 +669,8 @@ filenames = {
     "shell/common/language_util.h",
     "shell/common/logging.cc",
     "shell/common/logging.h",
+    "shell/common/mac_util.h",
+    "shell/common/mac_util.mm",
```
Comment (jkleinsc) [inline filenames.gni:673]: These should be added to lib_sources_mac: https://github.com/electron/electron/blob/afa1fdb2576551212d44609f681615e465878faa/filenames.gni#L122

**Review by jkleinsc — CHANGES_REQUESTED** [review]
LGTM, just needs a slight tweak in filenames.gni

File: filenames.gni (hunk 2 above)
Comment (ckerr) [reply filenames.gni]: @jkleinsc moved in cfc63651697ed25d18efd1d278026a10a48dc398

**Review by jkleinsc — APPROVED** [review]

**Pull request merged**